ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -65,104 +71,158 @@ def init_regex_and_parser(self, data_directory_path):
+        # create number_words dict having number variants and their corresponding scale and increment value
+        # create language_scale_map dict having scale variants and their corresponding value
+        numeral_df = pd.read_csv(os.path.join(data_directory_path, NUMBER_NUMERAL_CONSTANT_FILE), encoding='utf-8')
+        for index, row in numeral_df.iterrows():
+            name_variants = self._get_list_from_pipe_sep_string(row[NUMBER_DATA_FILE_NAME_VARIANTS])
             value = row[NUMBER_DATA_FILE_VALUE]
             if float(value).is_integer():
                 value = int(row[NUMBER_DATA_FILE_VALUE])
             number_type = row[NUMBER_DATA_FILE_TYPE]
 
-            if value in NUMBER_DIGIT_UNITS:
-                self.language_number_map[number] = value
-                self.language_number_map[str(value)] = value
-
-            numerals_list = self._get_numerals_list(numerals)
             if number_type == NUMBER_TYPE_UNIT:
-                self.numbers_word[number] = (1, value)
-                self.numbers_word[str(value)] = (1, value)
-                for numeral in numerals_list:
+                for numeral in name_variants:
                     self.numbers_word[numeral] = (1, value)
             else:
-                self.numbers_word[number] = (value, 0)
-                self.numbers_word[str(value)] = (value, 0)
-                for numeral in numerals_list:
+                for numeral in name_variants:
                     self.numbers_word[numeral] = (value, 0)
-
-    def _detect_number_from_numerals(self, number_list, original_list):
+                    self.language_scale_map[numeral] = value
+
+        # create units_dict having unit variants and their corresponding value
+        unit_file_path = os.path.join(data_directory_path, NUMBER_UNITS_FILE)
+        if os.path.exists(unit_file_path):
+            units_df = pd.read_csv(unit_file_path, encoding='utf-8')
+            for index, row in units_df.iterrows():
+                unit_variants = self._get_list_from_pipe_sep_string(row[NUMBER_UNIT_VARIANTS])
+                unit_value = row[NUMBER_UNIT_VALUE]
+                for unit in unit_variants:
+                    self.units_map[unit] = unit_value
+
+    def _get_unit_from_text(self, detected_original, processed_text):
+        """
+        Method to detect unit from detected original text.
+        Args:
+            detected_original (str): detected number text from processed text
+            processed_text (str): processed text
+        Returns:
+            unit(str|<None>): unit detected
+            original_text(str): original text adding unit text if detected
+        """
+        unit = None
+        original_text = detected_original
+
+        if not self.units_map:
+            return unit, original_text
+
+        unit_choices = "|".join(self.units_map)
+        unit_matches = re.findall(r'((' + unit_choices + r')?[\.\,\s]*?' + detected_original + r'\s*(' + unit_choices +
```

Comment:
`re.escape` detected_original

Reply:
Remove redundant ? in `[\.\,\s]*?`

`[\.\,\s]*?` => `[\.\,\s]*`

Reply:
Can use `re.search` and then unpack using 
`original_text, unit_prefix, unit_suffix = re.groups()` if a search is found

Reply:
Done